services/core/shared/core.js

```diff
@@ -355,6 +460,78 @@ const EMAcalc = (feePerByte, prevFeeEstPerByte) => {
 	return EMAoutput;
 };
 
+const getEstimateFeeByte = async () => {
+	const cacheKeyFeeEst = 'lastFeeEstimate';
+
+	const prevFeeEstPerByte = { blockHeight: config.feeEstimates.defaultStartBlockHeight };
+	const cachedFeeEstPerByte = await cacheRedisFees.get(cacheKeyFeeEst);
+	const latestBlock = await getBlocks({ sort: 'height:desc', limit: 1 });
+	if (cachedFeeEstPerByte
+		&& ['low', 'med', 'high', 'updated', 'blockHeight', 'blockId']
+			.every(key => Object.keys(cachedFeeEstPerByte).includes(key))) {
+		if ((Date.now() / 1000) - cachedFeeEstPerByte.updated < 10
+			|| Number(latestBlock.data.id) === cachedFeeEstPerByte.blockHeight) {
+			return cachedFeeEstPerByte;
+		}
+
+		Object.assign(prevFeeEstPerByte, cachedFeeEstPerByte);
+	}
+
+	const coreLogic = async (blockBatch, innerPrevFeeEstPerByte) => {
```

Comment:
shouldnt this function defined outside?

Reply:
This function isn't necessary for any other operation, other than computing the fee estimates. And, since it forms the core logic of the `getEstimateFeeByte` method as mentioned [here](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0016.md#at-the-node) in LIP-0016, we decided to keep it as an inner function.